### PR TITLE
Assets refactor round2

### PR DESCRIPTION
**Pull request by willianrampazzo**
This is the second round of the Assets class refactor based on comments made at #3411 

-  Move test_utils_assets.py to the correct selftests directory;
- Refactor _verify method;
- Move the json calls inside the `with` statements;
- Change EnvironmentError exception to OSError.

### CODE REVIEW

**Comment by willianrampazzo** [comment]
This PR is related to #3450.

File: avocado/utils/asset.py [hunk 1]
```diff
@@ -285,20 +284,17 @@ def _verify(self, asset_path):
         value as the hash of the asset_file, otherwise return False.
         :rtype: bool
         """
-        if not self.asset_hash:
```
Comment (apahim) [inline avocado/utils/asset.py]: This test should be `if self.asset_hash is None`. I think it was like that... not sure why it changed.

File: avocado/utils/asset.py [hunk 2]
```diff
@@ -285,20 +284,17 @@ def _verify(self, asset_path):
...
-            return True
-        if self._get_hash_from_file(asset_path) == self.asset_hash:
-            return True
-        else:
-            return False
+        return bool(not self.asset_hash or
+                    (self.asset_hash and
+                     self._get_hash_from_file(asset_path) == self.asset_hash))
```
Comment (apahim) [inline avocado/utils/asset.py]: Why is this change needed? I mean, this is way less readable and the second `self.asset_hash` check inside the `boll()` is redundant. I'd just remove the else (while keeping the final return), but if you insist in changing this, you should consider using [any()](https://docs.python.org/3.6/library/functions.html#any).

File: avocado/utils/asset.py (hunk 1 above)
Comment (willianrampazzo) [reply avocado/utils/asset.py]: Okay, I'll revert on v2.

File: avocado/utils/asset.py (hunk 2 above)
Comment (willianrampazzo) [reply avocado/utils/asset.py]: Okay, I'll just remove the `else` on v2.

**Comment by willianrampazzo** [comment]
Sending v2.
